src/ui/components/SearchInput/style.scss

```diff
@@ -0,0 +1,91 @@
+@import "~core/css/inc/mixins";
+@import "~ui/components/Icon/vars";
+
+$input-height: 48px;
+$icon-margin-start: 24px;
+$icon-margin-end: 12px;
+$transition-duration: 250ms;
+
+.SearchInput {
+  position: relative;
+  text-align: center;
+  width: 100%;
+
+  .Icon-magnifying-glass {
+    @include margin(
+      ($input-height - $default-icon-size) / 2,
+      $icon-margin-end,
+      ($input-height - $default-icon-size) / 2,
+      $icon-margin-start);
+
+    vertical-align: top;
+  }
+
+  .SearchInput-animation-icon {
+    @include start(0);
+
+    opacity: 0;
+    position: absolute;
+    transition: opacity 1ms $transition-duration, transform $transition-duration;
+  }
+}
+
+.SearchInput--text {
+  @include text-align-start();
+
+  .SearchInput-label {
+    color: transparent;
+
+    & .Icon-magnifying-glass {
+      opacity: 0;
+      transition: none;
+    }
+  }
+
+  .SearchInput-animation-icon {
+    color: #fff;
+    opacity: 1;
+    transform: translateX(0) !important;
```

Comment:
If this `!important` is really needed, a comment explaining why would be handy; it will surely prevent a refactoring future developer from trying to hunt its purpose down trying to remove it. 😉